pages/architecture/distributed_registry.rst

```diff
@@ -0,0 +1,230 @@
+  to share it with other instances. This helps to mitigate the risk of data loss.
+
+Security
+--------
+
+* Data in the registry should be digitally signed so that the identity of the producing party and the integrity
+  of the data can be reassured.
+
+Maintainability
+---------------
+
+* Changes should be backwards compatible so that new features don't break historical data.
+
+Roles
+*****
+
+===========================  ======================================================
+Role / Entity                Description
+===========================  ======================================================
+Care organisation            Healthcare organisation (e.g. hospital, insurance company,
+                             who wishes to exchange data with other care organisations
+                             through their respective vendors using Nuts.
+Care organisation signatory  Person within a care organisation who is authorised to
+                             sign in the name of their organisation.
+Vendor                       Organisation who supplies and operates a care organisation's
+                             software systems containing medical data.
+Nuts administrator           Person within the Nuts Foundation who is authorised to
+                             accept registrations from vendors.
+Node                         Vendor's system used to communicate with the network,
+                             conform the Nuts specification.
+===========================  ======================================================
+
+Architecture
+************
+
+The data contained in the registry is modelled as streams of events. Every stream starts with a genesis event,
+every successive event refers to its preceding event. Events are immutable, so every change (or deletion) is
+recorded as a new event.
+For now, only the registration of a new vendor starts a new stream. The minimal event stream leading to an
+endpoint looks as following:
+
+.. raw:: html
+    :file: ../../_static/images/distributed-registry-vendor-stream.svg
+
+Every event contains the following fields, aside from its event-specific data:
+
+======================  ===========  ====
+Field                   Format       Description
+======================  ===========  ====
+ID                      UUID / hash  Uniquely identifies this event
+Previous event ID       UUID / hash  Refers to the previous event in this stream
+Date of issue           timestamp    Timestamp the event was emitted
+Signature               TBD          Signature which authenticates the event
+======================  ===========  ====
+
+Processes
+*********
+
+Registering a vendor
+--------------------
+This process makes a vendor known to the network. Afterwards:
+
+* The vendor is known to the network.
+* The vendor van register their contracts with care organisations.
+
+Limitations:
+
+* For now, we only support 1 node per vendor.
```

Comment:
not necessarily, we'll have to check the CA tree and separate the concerns the correct way.